src/weights.jl

```diff
@@ -5,20 +5,24 @@ abstract type AbstractWeights{S<:Real, T<:Real, V<:AbstractVector{T}} <: Abstrac
     @weights name
 
 Generates a new generic weight type with specified `name`, which subtypes `AbstractWeights`
-and stores the `values` (`V<:RealVector`) and `sum` (`S<:Real`).
+and stores the `values` (`V<:RealVector`), the pre-computed `sum` (`S<:Real`) and
+whether all values are `positive`.
 """
 macro weights(name)
     return quote
         mutable struct $name{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractWeights{S, T, V}
             values::V
             sum::S
-            function $(esc(name)){S, T, V}(values, sum) where {S<:Real, T<:Real, V<:AbstractVector{T}}
+            positive::Union{Bool, Missing}
```

Comment:
It seems it's actually used to store whether all elements are non-negative rather than positive?

Reply:
Well, yeah, that's "positive", not "strictly positive". But maybe "nonnegative" is more explicit. It's kind of ugly though. :-)

Reply:
It is internal so I would prefer a precise name.

Reply:
OK. Actually to avoid the double negation I propose simply calling this `negative` and reversing the logic.